traffic_ops/app/db/reencrypt/reencrypt.go

```
@@ -0,0 +1,406 @@
+	if err != nil {
+		fmt.Println("unmarshalling '", PROPERTIES_FILE, "': ", err.Error())
+		os.Exit(0)
+	}
+
+	sslStr := "require"
+	if !pgCfg.SSL {
+		sslStr = "disable"
+	}
+	db, err := sqlx.Open("postgres", fmt.Sprintf("postgres://%s:%s@%s:%d/%s?sslmode=%s&fallback_application_name=trafficvault", pgCfg.User, pgCfg.Password, pgCfg.Hostname, pgCfg.Port, pgCfg.DBName, sslStr))
+	if err != nil {
+		fmt.Println("opening database: " + err.Error())
+		os.Exit(0)
+	}
+
+	if err = reEncryptSslKeys(db, previousKey, newKey); err != nil {
+		fmt.Println("re-encrypting SSL Keys: ", err.Error())
+		os.Exit(0)
+	}
+	if err = reEncryptUrlSigKeys(db, previousKey, newKey); err != nil {
+		fmt.Println("re-encrypting URL Sig Keys: ", err.Error())
+		os.Exit(0)
+	}
+	if err = reEncryptUriSigningKeys(db, previousKey, newKey); err != nil {
+		fmt.Println("re-encrypting URI Signing Keys: ", err.Error())
+		os.Exit(0)
+	}
+	if err = reEncryptDNSSECKeys(db, previousKey, newKey); err != nil {
+		fmt.Println("re-encrypting DNSSEC Keys: ", err.Error())
+		os.Exit(0)
+	}
+
+	if err = updateKeyFile(previousKey, *previousKeyLocation, newKey); err != nil {
+		fmt.Println("updating the key file: ", err.Error())
+		os.Exit(0)
+	}
+
+	fmt.Println("Successfully re-encrypted keys for SSL Keys, URL Sig Keys, URI Signing Keys, and DNSSEC Keys.")
+}
+
+type Config struct {
+	DBName   string `json:"dbname"`
+	Hostname string `json:"hostname"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	Port     int    `json:"port"`
+	SSL      bool   `json:"ssl"`
+}
+
+func readKey(keyLocation string) ([]byte, error) {
+	var keyBase64 string
+	keyBase64Bytes, err := ioutil.ReadFile(keyLocation)
+	if err != nil {
+		return []byte{}, errors.New("reading file '" + keyLocation + "':" + err.Error())
+	}
+	keyBase64 = string(keyBase64Bytes)
+
+	key, err := base64.StdEncoding.DecodeString(keyBase64)
+	if err != nil {
+		return []byte{}, errors.New("AES key cannot be decoded from base64")
```

This should return the error as well, I think, to help diagnose the problem